src/main/tools/build-runfiles-windows.cc

```diff
@@ -12,16 +12,389 @@
+            (metadata.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY) != 0;
+        bool is_symlink =
+            (metadata.dwFileAttributes & FILE_ATTRIBUTE_REPARSE_POINT) != 0;
+        if (is_symlink) {
+          wstring target;
+          if (!blaze_util::ReadSymlinkW(subpath, &target)) {
+            die(L"ReadSymlinkW failed (%s): %hs",
+                subpath.c_str(), GetLastErrorString().c_str());
+          };
+
+          subpath = AsAbsoluteWindowsPath(subpath.c_str());
+          target = AsAbsoluteWindowsPath(target.c_str());
+          ManifestFileMap::iterator expected_target =
+              manifest_file_map.find(subpath);
+
+          if (expected_target == manifest_file_map.end()
+              || expected_target->second.empty()
+              || target != AsAbsoluteWindowsPath(expected_target->second.c_str())
+              || blaze_util::IsDirectoryW(target) != is_dir) {
+            if (is_dir) {
+              RemoveDirectoryOrDie(subpath);
+            } else {
+              DeleteFileOrDie(subpath);
+            }
+          } else {
+            manifest_file_map.erase(expected_target);
+          }
+        } else {
+          if (is_dir) {
+            ScanTreeAndPrune(subpath);
+            // If the directory is empty, then we remove the directory.
+            // Otherwise RemoveDirectory will fail with ERROR_DIR_NOT_EMPTY,
+            // which we can just ignore.
+            if (!RemoveDirectoryW(subpath.c_str())
+                && GetLastError() != ERROR_DIR_NOT_EMPTY) {
+              die(L"RemoveDirectoryW failed (%s): %hs",
+                  subpath.c_str(), GetLastErrorString().c_str());
+            }
+          } else {
+            DeleteFileOrDie(subpath);
+          }
+        }
+      }
+    } while (::FindNextFileW(handle, &metadata));
+    ::FindClose(handle);
+  }
+
+  void CreateFiles(bool creating_symlink_needs_admin_privilege) {
+    DWORD privilege_flag = creating_symlink_needs_admin_privilege
+                           ? 0
+                           : SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE;
+
+    for (ManifestFileMap::const_iterator it = manifest_file_map.begin();
+         it != manifest_file_map.end(); ++it) {
+
+      // Ensure the parent directory exists
+      wstring parent_dir = GetParentDirFromPath(it->first);
+      if (!DoesDirectoryPathExist(parent_dir.c_str())) {
+        MakeDirectoriesOrDie(parent_dir);
+      }
+
+      if (it->second.empty()) {
+        // Create an empty file
+        HANDLE h = CreateFileW(it->first.c_str(),     // name of the file
+                               GENERIC_WRITE, // open for writing
+                               0,            // sharing mode, none in this case
+                               0,            // use default security descriptor
+                               CREATE_ALWAYS, // overwrite if exists
+                               FILE_ATTRIBUTE_NORMAL,
+                               0);
+        if (h) {
```

Comment:
`CreateFileW` returns `INVALID_HANDLE_VALUE` on error. Please compare `h` to that instead of to `0`.